BitwardenShared/Core/Platform/Services/AuthenticatorSyncServiceTests.swift

```diff
@@ -455,6 +455,31 @@ final class AuthenticatorSyncServiceTests: BitwardenTestCase { // swiftlint:disa
         waitFor(!errorReporter.errors.isEmpty)
     }
 
+    /// The AuthService may not get notified about Vault locking if the user has switched accounts. Verify
+    /// that it checks the vault lock before beginning to decrypt and write new Ciphers received.
+    ///
+    @MainActor
+    func test_writeCiphers_vaultLocked() async throws {
+        setupInitialState()
+        await subject.start()
+        stateService.syncToAuthenticatorSubject.send(("1", true))
+        try await Task.sleep(nanoseconds: 10_000_000)
```

Comment:
Yeah, the sleeps that are in these tests are because there isn't an explicit thing to wait _for_. We need to give the service time to do things like subscribe to Ciphers or something else that happens out of the view of the tests. So we're waiting to avoid testing the wrong thing or false positives. 

In this case, not waiting can cause us to set the vault to locked before the `syncToAuthenticatorSubject.send` kicks off the `determineSyncForUserId` method. That would mean we look at the vault status and (correctly) don't set anything up. But in this case, we actually want to test that _after_ we've subscribed to cipher updates, we get the vault locked without a notification. So we have to wait until that subscription happens but there's not an external thing the tests can wait for to know it happens. I actually tested this one using code coverage and breakpoints to see that when I wrote it without any sleep, it never gets into the guard I'm trying to verify.